### ISSUE TITLE
CMD+F toggle

### ISSUE THREAD

**Issue by tumispro** [jiahaog/nativefier#29]
It would be cool if you could enable page search using CMD + F for apps. Would be very useful for wikis.

**Comment by jiahaog** [comment]
Electron does not support this, but it's possible with [`window.find()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/find).

**Comment by sevenam** [comment]
+1 for this.

Find is something I use a lot in most web apps I use.

**Comment by RonnyO** [comment]
+1, I miss it on some apps

**Comment by brunorsini** [comment]
+1, really miss this : /

**Comment by swrobel** [comment]
@jiahaog any suggestions for how to use `window.find`? I can get it to return `true` like so but it doesn't scroll or highlight or do anything useful:
```javascript
> window.find('union')
true
```

**Comment by jiahaog** [comment]
I think you can use [webContents.findInPageText()](https://github.com/atom/electron/blob/master/docs/api/web-contents.md#webcontentsfindinpagetext-options)!

**Comment by swrobel** [comment]
```javascript
> webContents.findInPage('union')
Uncaught ReferenceError: webContents is not defined(…)
```

**Comment by swrobel** [comment]
Maybe you need to use [inspectServiceWorker()](http://electron.atom.io/docs/v0.35.0/api/web-contents/#webcontentsinspectserviceworker) instead of [toggleDevTools()](https://github.com/jiahaog/nativefier/blob/master/app/src/components/menu/menu.js#L148)?

**Comment by jb510** [comment]
I've been using a couple webapps (Asana, Harvest, Slack) wrapped in nativefier for a couple months now.  I'm surprised how frequently (dozens of times per day) I try/need to use cmd-F to find something on the page and can't.

**Comment by brunorsini** [comment]
+1000 :)

**Comment by swrobel** [comment]
http://electron.atom.io/docs/api/web-contents/#contentsfindinpagetext-options

**Comment by azu** [comment]
https://github.com/nakajmg/electron-search-text

**Comment by swrobel** [comment]
This project looks abandoned, but if somehow it isn't, here's a ready-to-use electron package to add search in page: https://github.com/rhysd/electron-in-page-search

**Comment by brunorsini** [comment]
I'd LOVE this, its lack breaks many of my workflows : /

**Comment by binyominzeev** [comment]
Hi, I tried Nativefier in Linux and would be great to use the search. Also installed the link above by @swrobel, and tried to compile the app again, but nothing changed.

How to use this in practice?

Wouldn't it make sense to have it compiled into Nativefier, without any external tools? (Simply put: why is this issue closed?)

**Comment by ronjouch** [comment]
> *"Wouldn't it make sense to have it compiled into Nativefier, without any external tools? (Simply put: why is this issue closed?)"*

@binyominzeev the feature would make total sense, yes, PR welcome. We adjusted our triage rules, and indeed this issue shouldn't be closed, thanks for pointing it out 👍, re-opening.

**Comment by brunorsini** [comment]
Today alone I missed this feature on 3 different occasions...

**Comment by basu90** [comment]
+1

**Comment by ronjouch** [comment]
Yup it's a desired feature, everybody knows it and the upvotes on the issue make it clear. Locking conversation to avoid "+1" comments; PR welcome 🙂.

**Comment by TheCleric** [comment]
I am active developing a PR for this issue if anyone would like to help.

I've got search about half working at the moment (get a search dialog to show up, performing a search, getting a match count) but running into a few snags (search box losing focus on input event, search matches no highlighting like they were when I began testing this at a lower level).

Right now I'm moving the search to it's own window like this project as there is a bit of overlap between that sample and what I'm trying to accomplish:

https://discuss.atom.io/t/using-webcontents-findinpage-to-implement-typical-find-function/25309/2

**Comment by alecmev** [comment]
[`electron-find`](https://github.com/TheoXiong/electron-find) might be of interest.

**Comment by eponymous301** [comment]
I took a look at `electron-find` and the docs for `--inject` and started experimenting but couldn't figure out how to get `electron-find` to run as an injected script. 

Are there any examples available of injected JS that accesses electron? I looked around but could not find anything other than trivial 'alert()' examples.

Below are a couple things I tried:

Minimal electron-find example, bundled with Browserify into a single .js file. (crapped out with: ` fs.existsSync is not a function`, in function `getElectronPath ()`)
```
const { remote, globalShortcut } = require("electron");
const { FindInPage }  = require("electron-find");

const findInPage = new FindInPage(remote.getCurrentWebContents());
globalShortcut.register('Command+f', findInPage.openFindWindow);
```

After poking around in dev console, ran the following to try to figure out how to access electron from the injected js:
```
const preLoad = module.parent;
const children = preLoad.children;
const electron = children.filter(x => x.id === "electron")[0];

const {ipcRenderer} = electron.exports;
console.log("IPC RENDERER");
console.log(ipcRenderer);

const {remote} = require("electron");
console.log("REMOTE");
console.log(remote);

const {globalShortcut} = electron.exports;
console.log("GLOBAL SHORTCUT");
console.log(globalShortcut);
```
and wound up with below output in dev console - I found `ipcRenderer`  but not `remote` or `globalShortcut`:
<img width="557" alt="Screen Shot 2020-12-23 at 4 21 24 PM" src="https://user-images.githubusercontent.com/29875759/103046068-b3fcac80-453b-11eb-9cc6-79e47cff998b.png">

**Comment by ronjouch** [comment]
@eponymous301 see our [Development guide](https://github.com/jiahaog/nativefier/blob/master/docs/development.md) to get yourself started hacking on Nativefier. This is not the job of an injected script.
